### PR TITLE
[FEAT] removes the collectible key if there are none of the type placed

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR removes the collectible key in the collectibles object if there are none of that type placed. 

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tests

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/goblins/bank/lib/bankUtils.ts [hunk 1]
```diff
@@ -62,7 +63,10 @@ export function canWithdraw({ item, game }: CanWithdrawArgs) {
   }
 
   // Placed items
-  if (item in game.collectibles) {
+  if (
+    item in game.collectibles &&
+    game.collectibles[item as CollectibleName]?.length
```
Comment (spencerdezartsmith) [inline src/features/goblins/bank/lib/bankUtils.ts:68]: Add in the check for keys with an empty array value currently. This will avoid a migration.

**Review by c-py — APPROVED** [review]
Awesome mate, LGTM!

**Comment by Myth727** [comment]
This version 6.30 has not fixed the problem for me yet on withdrawal of item's.  Now my nfts dug up show up as selectable on the withdraw screen.  But when going forth I get the crash screen saying there's an error.

**Comment by Myth727** [comment]
Screenshot_20221128-194212_MetaMask

Get this error when trying to make a withdrawal.

**Pull request merged**